# 00 · Data Check

**Goal:** Load the Enron-Spam dataset and get first feel for it before doing anything else — how many emails, how the labels are balanced, and where the missing values are.I want to know what I'm working with before I start cleaning or modeling.

In [2]:
%pip install pandas


     |████████████████████████████████| 11.6 MB 1.0 MB/s eta 0:00:01    |█████▊                          | 2.1 MB 4.0 MB/s eta 0:00:03
     |████████████████████████████████| 21.2 MB 5.6 MB/s eta 0:00:011
You should consider upgrading via the '/Library/Frameworks/Python.framework/Versions/3.9/bin/python3 -m pip install --upgrade pip' command.
Note: you may need to restart the kernel to use updated packages.


In [3]:
import pandas as pd

df = pd.read_csv('../data/enron_spam_data.csv')
df.shape

(33716, 5)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33716 entries, 0 to 33715
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Message ID  33716 non-null  int64 
 1   Subject     33427 non-null  object
 2   Message     33345 non-null  object
 3   Spam/Ham    33716 non-null  object
 4   Date        33716 non-null  object
dtypes: int64(1), object(4)
memory usage: 1.3+ MB


In [5]:
df.head()

,Message ID,Subject,Message,Spam/Ham,Date
0,0,christmas tree farm pictures,NaN,ham,1999-12-10
1,1,"vastar resources , inc .","gary , production from the high island larger ...",ham,1999-12-13
2,2,calpine daily gas nomination,- calpine daily gas nomination 1 . doc,ham,1999-12-14
3,3,re : issue,fyi - see note below - already done .\nstella\...,ham,1999-12-14
4,4,meter 7268 nov allocation,fyi .\n- - - - - - - - - - - - - - - - - - - -...,ham,1999-12-14


In [6]:
df['Spam/Ham'].value_counts()

Spam/Ham
spam    17171
ham     16545
Name: count, dtype: int64

In [7]:
df['Spam/Ham'].value_counts(normalize=True).round(3)

Spam/Ham
spam    0.509
ham     0.491
Name: proportion, dtype: float64

In [8]:
df.isnull().sum()

Message ID      0
Subject       289
Message       371
Spam/Ham        0
Date            0
dtype: int64

## First check summary

- 33,716 emails total, with 5 columns (Message ID, Subject, Message, Spam/Ham, Date)
- Labels are balanced: about 51% spam, 49% ham
- No missing labels, so the data is ready for supervised learning
- A few missing values: 289 empty subjects, 371 empty messages (will fill with empty strings later)